common.mk

```diff
@@ -177,32 +177,30 @@ endif
 		--annotation-file $(FINAL_ANNO_FILE) \
 		--log-level $(FIRRTL_LOGLEVEL) \
 		--allow-unrecognized-annotations \
-	  $(REPL_SEQ_MEM) \
-		$(HARNESS_CONF_FLAGS) \
 		-X $(SFC_LEVEL) \
 		$(EXTRA_FIRRTL_OPTIONS))
-	-mv $(SFC_FIRRTL_BASENAME).mid.fir $(SFC_FIRRTL_FILE)
-	-mv $(SFC_FIRRTL_BASENAME).lo.fir $(SFC_FIRRTL_FILE)
-	$(if $(ENABLE_CUSTOM_FIRRTL_PASS), cat $(SFC_ANNO_FILE) | jq 'del(.[] | select(.target | test("io.cpu"))?)' > $(build_dir)/tmp.json,)
-	$(if $(ENABLE_CUSTOM_FIRRTL_PASS), cat $(build_dir)/tmp.json > $(SFC_ANNO_FILE) && rm $(build_dir)/tmp.json,)
+	-mv $(SFC_FIRRTL_BASENAME).lo.fir $(SFC_FIRRTL_FILE) # Optionally change file type when SFC generates LowFIRRTL
+	@if [ "$(SFC_LEVEL)" = low ]; then cat $(SFC_ANNO_FILE) | jq 'del(.[] | select(.target | test("io.cpu"))?)' > /tmp/unnec-anno-deleted.sfc.anno.json; fi
+	@if [ "$(SFC_LEVEL)" = low ]; then cat /tmp/unnec-anno-deleted.sfc.anno.json > $(SFC_ANNO_FILE) && rm /tmp/unnec-anno-deleted.sfc.anno.json; fi
 	firtool \
 		--format=fir \
-		-O=release \
 		--dedup \
 		--export-module-hierarchy \
 		--emit-metadata \
-		--verify-each=false \
+		--verify-each=true \
+		--warn-on-unprocessed-annotations \
 		--disable-annotation-classless \
 		--disable-annotation-unknown \
-		--warn-on-unprocessed-annotations \
+		--mlir-timing \
 		--lowering-options=emittedLineLength=2048,noAlwaysComb,disallowLocalVariables,explicitBitcast,verifLabels,locationInfoStyle=wrapInAtSquareBracket \
 		--repl-seq-mem \
-		--repl-seq-mem-circuit=$(MODEL) \
 		--repl-seq-mem-file=$(FIRTOOL_SMEMS_CONF) \
+		--repl-seq-mem-circuit=$(MODEL) \
 		--annotation-file=$(SFC_ANNO_FILE) \
 		--split-verilog \
 		-o $(OUT_DIR) \
 		$(SFC_FIRRTL_FILE)
+	-mv $(SFC_SMEMS_CONF) $(FIRTOOL_SMEMS_CONF)
```

Comment:
Is this the correct location for this `mv`? Wouldn't the SFC smems file override the FIRTOOL one (if they both generate an smems file)? Or are you implicitly depending that either FIRTOOL or SFC generate the file separately and never at the same time?

Reply:
IIRC `SFC_SMEMS_CONF` is only generated in the SFC-LowFIRRTL case, while `FIRTOOL_SMEMS_CONF` is generated only in the SFC-CHIRRTL case. So this implicitly assumes that the smems file is generated by by either the SFC or FIRTOOL, not both.